runtime/near-vm-logic/tests/test_promises.rs

```diff
@@ -0,0 +1,286 @@
+    let config = Config::default();
+    let promise_results = vec![];
+    let mut memory = MockedMemory::default();
+    let mut logic = VMLogic::new(&mut ext, context, &config, &promise_results, &mut memory);
+    let index = promise_create(&mut logic, b"rick.test", 0, 0).expect("should create a promise");
+    let code = b"sample";
+
+    logic
+        .promise_batch_action_deploy_contract(123, code.len() as u64, code.as_ptr() as _)
+        .expect_err("shouldn't accept not existent promise index");
+    let non_receipt = logic
+        .promise_and(index.to_le_bytes().as_ptr() as _, 1u64)
+        .expect("should create a non-receipt promise");
+    logic
+        .promise_batch_action_deploy_contract(non_receipt, code.len() as u64, code.as_ptr() as _)
+        .expect_err("shouldn't accept non-receipt promise index");
+
+    logic
+        .promise_batch_action_deploy_contract(index, code.len() as u64, code.as_ptr() as _)
+        .expect("should add an action to deploy contract");
+    assert_eq!(logic.used_gas().unwrap(), 540);
+    let expected = "[{\"receipt_indices\":[],\"receiver_id\":\"rick.test\",\"actions\":[{\"FunctionCall\":{\"method_name\":\"promise_create\",\"args\":\"args\",\"gas\":0,\"deposit\":0}},{\"DeployContract\":{\"code\":[115,97,109,112,108,101]}}]}]";
+    assert_eq!(&serde_json::to_string(ext.get_receipt_create_calls()).unwrap(), &expected);
+}
+
+#[test]
+fn test_promise_batch_action_transfer() {
+    let mut ext = MockedExternal::default();
+    let mut context = get_context(vec![]);
+    context.account_balance = 100;
+    context.attached_deposit = 10;
+    let config = Config::default();
+    let promise_results = vec![];
+    let mut memory = MockedMemory::default();
+    let mut logic = VMLogic::new(&mut ext, context, &config, &promise_results, &mut memory);
+    let index = promise_create(&mut logic, b"rick.test", 0, 0).expect("should create a promise");
+
+    logic
+        .promise_batch_action_transfer(123, 110u128.to_le_bytes().as_ptr() as _)
+        .expect_err("shouldn't accept not existent promise index");
+    let non_receipt = logic
+        .promise_and(index.to_le_bytes().as_ptr() as _, 1u64)
+        .expect("should create a non-receipt promise");
+    logic
+        .promise_batch_action_transfer(non_receipt, 110u128.to_le_bytes().as_ptr() as _)
+        .expect_err("shouldn't accept non-receipt promise index");
+
+    logic
+        .promise_batch_action_transfer(index, 110u128.to_le_bytes().as_ptr() as _)
+        .expect("should add an action to transfer money");
+    logic
+        .promise_batch_action_transfer(index, 1u128.to_le_bytes().as_ptr() as _)
+        .expect_err("not enough money");
+    assert_eq!(logic.used_gas().unwrap(), 440);
+    let expected = "[{\"receipt_indices\":[],\"receiver_id\":\"rick.test\",\"actions\":[{\"FunctionCall\":{\"method_name\":\"promise_create\",\"args\":\"args\",\"gas\":0,\"deposit\":0}},{\"Transfer\":{\"deposit\":110}}]}]";
```

Comment:
Use `serde_json::json` to make stuff like this more readable https://docs.serde.rs/serde_json/macro.json.html